src/docs/patterns/user-auth.mdx

```diff
@@ -0,0 +1,296 @@
+
+-   Enter email
+-   Success (email has been sent)
+-   Receive an email for resetting the password
+-   Create password
+-   Success & welcome
+
+<ImageGrid
+    images={[ForgotPasswordEnterEmail, ForgotPasswordConfirmSent, ForgotPasswordNewPassword, ForgotPasswordWelcome]}
+    caption={['1: Enter email', '2: Success (Email has been sent)', '3: Create Password', '4: Success & Welcome']}
+/>
+
+**Change password**
+
+When the users are logged in, they have the option to reset their password by going through the changing password workflow. This workflow can usually be triggered from user preference settings. The workflow involves the following screens:
+
+-   Change password
+-   Success & welcome
+
+<React.Fragment>
+    <ImageGrid
+        images={[ChangePassword, ChangePasswordWelcome]}
+        caption={['1: Change password', '2: Success & welcome']}
+    />
+    <Divider />
+</React.Fragment>
+
+# Behavior
+
+## Email Communications
+
+<!-- This might deserve its own pattern. We already have something in the backlog to design for an email template. -->
+
+Email communications are frequently used throughout the user authentication workflow since they typically involve some security changes. You should never give the users option to unsubscribe from account-related email communications.
+
+<ImageGrid
+    images={[EmailCommunicationInvitation, EmailCommunicationVerification]}
+    caption={['Two examples of email communications', '1: Invitation to join', '2: Verifying email address']}
+/>
+
+## Success & Welcome Screen
+
+<!-- Welcome screen might deserve to be its own pattern (or at least merge into something like "onboarding"). -->
+<!-- TODO: add more images here, if studio blue can release more. -->
+
+The purpose of a welcome screen is to assure the user that something has been indeed achieved, and thus can be more delightful. You are encouraged to use animations and [graphics](/style/illustrations).
+
+## Terminated Workflows & Expired Sessions
+
+During the registration, if a user exits the application before they ever create a password for their account, the whole session is treated as expired, and the users should start over again.
+
+If an invitation link or verification code has expired, the users should be able to request for another invitation link or verification code.
+
+## Security Concerns
+
+### Error Messages
+
+When a user encounters an error, whether it is a password mismatch or the server is temporarily down, they should be notified with an appropriate error message. However, be very careful when presenting error messages to the users. You should only show what is necessary to recover from the problem, such as "incorrect email or password" and "server unavailable".
+
```

Comment:
the show what's necessary (error message) shouldn't indicate incorrect email or incorrect password. Can we use some of the verbiage from original login doc for this section.

If authentication fails, the user remains on the login page and the screen should be updated to show an appropriate error message (e.g., Invalid Credentials).

If an error occurs when attempting to validate the credentials (e.g., Server Unavailable), the user should be notified with an appropriate error message.

- NOTE: be very careful when presenting error messages to users. For security, you should not present more information than is necessary (e.g., do NOT provide messages like "User does not exist," "Incorrect Password," etc.).

Reply:
@jeffvg @joebochill may you guys clarify what exactly shall I change?

Reply:
for this section - You should only show what is necessary to recover from the problem, such as "incorrect email or password" and "server unavailable".

maybe something like
- You should only show what is necessary to recover from the problem, such as "invalid credentials and "server unavailable".

NOTE: be very careful when presenting error messages to users. For security, you should not present more information than is necessary (e.g., do NOT provide messages like "User does not exist," "Incorrect Password," etc.).

Reply:
@jeffvg "incorrect email or password" is a phrase I grabbed from WAS. Would you suggest me to change that to "invalid credentials", in the text and in the illustrations?

Reply:
I would say yes. Normally we wouldn't want to call out what part of the credentials are incorrect. If password is incorrect, both ID and password fields would error "invalid credentials". Keeping hackers away....lol